controllers/events.js

```diff
@@ -1,45 +1,77 @@
-const db = require("../config/database");
-const getAllEvents = require("../services/tmQueryService/tmEventQueries.js");
+const User = require("../models/User.js");
+const Event = require("../models/Event.js");
+const getNewEvents = require("../services/tmQueryService/tmEventQueries.js");
 // require the Ticketmaster API call service functions
 
 const newEvents = async (req, res) => {
   try {
-    console.log("made it here");
-    // req.body should have the location data of the users (Lat and Long)
-    // call services / Ticketmaster API to get a list of events back based on the user location
-    const newTMEventsData = await getAllEvents(req.body.location);
-    console.log(newTMEventsData.data);
-    // ["_embedded"][
-    //   "events"
-    // ]; // an array of TM event objects
-
-    const response = newTMEventsData.data._embedded.events;
-
-    const newEvents = [];
-
-    // newTMEventsData.forEach((newTMEvent) => {
-    //   newEvents.push({
-    //     eventID: newTMEvent.id,
-    //     priceRange: newTMEvent.priceRanges[0].min,
-    //     dates: newTMEvent.dates.start.localDate,
-    //     eventURL: newTMEvent.url,
-    //     description: newTMEvent.info,
-    //     classifications: newTMEvent.classifications[0].segment.name,
-    //     venue: {
-    //       location: newTMEvent["_embedded"].venues[0].location, // do we need this attribute? it's a latitude & longitude object
-    //       name: newTMEvent["_embedded"].venues[0].name,
-    //       address: newTMEvent["_embedded"].venues[0].address,
-    //     },
-    //     // can we add image url to Events schema?
-    //   });
-    // });
-
-    // const addedEvents = await db.Events.insertMany(newEvents);
+    const foundUser = await User.findById(req.user.id)
+    
+    const newLocation = `${req.body.lat} ${req.body.lng}` // formatting the Location data to match User model
+
+    if (foundUser.currentLatLng !== newLocation || foundUser.eventQueue.length < 5) { // Only need to proceed if the Location has changed or the queue is too short:
+
+      //console.log('calling ticketmaster')
+
+      await User.findByIdAndUpdate(foundUser._id, {currentLatLng: newLocation}) // save the new location to the user < this is not working
+
+      const newTMEventsData = await getNewEvents({...req.body});
+
+      const eventsForQueue = newTMEventsData.data._embedded.events;
+
+      for (let i = 0; i < eventsForQueue.length; i++) {
```

Comment:
I was thinking maybe a check happen to see if anything good was returned from line 20 before this happens

Reply:
Yeah I created a separate issue for paging through the ticketmaster results,  I think what we would do is actually check at the end and if everything got skipped retry with a new page of results from ticketmaster.